validator_test.go

```diff
@@ -199,13 +351,114 @@ func TestValidatorOverwriteEmailListDirectly(t *testing.T) {
 	})
 	<-updated
 
-	if validator("xyzzy@example.com") {
-		t.Error("email removed from list should not validate")
+	for _, tc := range testCasesPostUpdate {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+			authorized := validator(tc.email)
+			g.Expect(authorized).To(Equal(tc.expectedAuthZ))
+		})
 	}
-	if !validator("plugh@example.com") {
-		t.Error("email retained in list should validate")
+}
+
+func TestValidatorSubDomains(t *testing.T) {
+	testCases := []struct {
+		name           string
+		allowedEmails  []string
+		allowedDomains []string
+		email          string
+		expectedAuthZ  bool
+	}{
+		{
+			name:           "EmailNotInCorrect1stSubDomainsNotInEmails",
+			allowedEmails:  []string{"xyzzy@example.com", "plugh@example.com"},
+			allowedDomains: []string{".example0.com", ".example1.com"},
+			email:          "foo.bar@example0.com",
+			expectedAuthZ:  false,
```

Comment:
This makes me feel like we need to update the documentation of this, I had expected that if I wrote `.example.com` that emails to that domain and its subdomains would have been allowed, this is how our whitelist domain works IIRC 🤔 

Secondly, do we have a test case where we do `.example.com` as an allowed domain and the email itself is `something@fooexample.com`, just to make sure that we are including the `.` in the suffix check and don't break that in the future by trying to be clever about something

Reply:
Good catch! You are right -- in `whitelistDomains` it does have a check for `redirectHostname == strings.TrimLeft(domain, ".")` essentially.

We should add that here to match.

Reply:
> This makes me feel like we need to update the documentation of this, I had expected that if I wrote `.example.com` that emails to that domain and its subdomains would have been allowed, this is how our whitelist domain works IIRC 🤔

To me if I write as allowed domain `.example.com` I expect all sub-domains to be allowed and not the main domain. If I write `example.com` I expect only the domain to be allowed (considering that we can't use wildcards). The way it is now implemented it allows the end-user to say allow any sub-domain but not the main domain (if you want to include the main domain the user can add it in the list). If we go with the behaviour similar to the whitelist I think we loose this use-case.

> Secondly, do we have a test case where we do `.example.com` as an allowed domain and the email itself is `something@fooexample.com`, just to make sure that we are including the `.` in the suffix check and don't break that in the future by trying to be clever about something

Just add a new test case for the suggested scenario.